sway/config.c

```diff
@@ -620,23 +621,127 @@ bool load_include_configs(const char *path, struct sway_config *config) {
 	return true;
 }
 
+/***************************************************************************//**
+*
+*	\brief	This function corrects the given 'line' if the opening parentheses
+*	happens to be on the next line.
+*
+*	\param[in]	file The pointer to the already open file.
+* \param[in/out]	line The current line read from the configuration file.
+* \param[in]	line_number Counter for the line number. Optional.
+*
+*	\return	A string to the corrected 'line' is returned.
+*
+*******************************************************************************/
+static inline char* handle_parentheses(FILE* const file, char* line,
+		unsigned int* const line_number) {
+	if (file && line) {
+		const size_t line_length = strlen(line);
+		/* If the last character of the line is not '{', we have to look for it on
+		 * the next line. */
+		if (1 < line_length && line[line_length-2u] != '{') {
+			char* next_line = NULL;
+			while (!feof(file) && !next_line) {
+				/* Read the file. Skip empty lines. Comments are skipped in read_line()
+				 * already. */
+				next_line = read_line(file);
+				if (line_number) {
+					++(*line_number);
+				}
+				if (next_line && is_empty(next_line)) {
+					free(next_line);
+					next_line = NULL;
+					continue;
+				}
+			}
+			if (next_line) {
+				if (next_line[0] == '{') {
+					if (!is_empty(next_line+1)) {
+						/* Currently we do not support something like 'mode "resize" {[…]}'
+						 * in one line but we should at least give the user an indication
+						 * what is wrong because that behavior is generally not expected. */
+						if (line_number) {
+							sway_log(L_ERROR, "Warning: Config (%u): Move additional commands to the "
+								"next line after '{'.", *line_number);
+						} else {
+							sway_log(L_ERROR, "Warning: Config: Move additional commands to the next "
+								"line after '{'.");
+						}
+					}
+
+					/* If we find the opening parenthesis on the next line, we append it
+					 * to 'line'. */
+					const size_t new_size = strlen(line) + strlen(next_line);
+					char* concat_line = malloc(new_size);
+					if (concat_line) {
+						line[strlen(line)-1] = '\0';  /* Override '\n' with '\0'. */
+						snprintf(concat_line, new_size, "%s %s", line, next_line);
+						free(line);
+						free(next_line);
+						line = concat_line;
+						next_line = NULL;
+					} else {
+						sway_log(L_ERROR, "Error: get_config_line(): Allocation of %zu bytes "
+							"was not successful.", new_size);
+					}
+				}
+			}
+		}
+	}
+	return line;
+}
```

Comment:
To be honest, rather than take this approach, I would prefer to make the whole config parser character oriented instead of line oriented, and make it a state machine, like a more typical parser would look. It would be a bit more complicated, but less sloppy and more things would work. It's my fault for not designing the config parser like this to begin with, for the record.

@minus7 might be interested in making that change. You can even make it syncronous if you like!